plugins/modules/ec2_asg_scheduled_action.py

```diff
@@ -0,0 +1,329 @@
+  register: scheduled_action
+
+- name: Create a scheduled action for autoscaling group
+  community.aws.ec2_asg_scheduled_action:
+    region: eu-west-1
+    autoscaling_group_name: test_asg
+    scheduled_action_name: test_scheduled_action
+    start_time: 2021 October 25 08:00 UTC
+    end_time: 2021 October 25 08:00 UTC
+    time_zone: Europe/London
+    recurrence: 40 22 * * 1-5
+    min_size: 10
+    max_size: 15
+    desired_capacity: 10
+    state: present
+  register: scheduled_action
+
+- name: Delete scheduled action
+  community.aws.ec2_asg_scheduled_action:
+    region: eu-west-1
+    autoscaling_group_name: test_asg
+    scheduled_action_name: test_scheduled_action
+    state: absent
+'''
+RETURN = r'''
+scheduled_action_name:
+  description: The name of the scheduled action.
+  returned: when I(state=present)
+  type: str
+  sample: test_scheduled_action
+start_time:
+  description: Start time for the action.
+  returned: when I(state=present)
+  type: str
+  sample: '2021 October 25 08:00 UTC'
+end_time:
+  description: End time for the action.
+  returned: when I(state=present)
+  type: str
+  sample: '2021 October 25 08:00 UTC'
+time_zone:
+  description: The ID of the Amazon Machine Image used by the launch configuration.
+  returned: when I(state=present)
+  type: str
+  sample: Europe/London
+recurrence:
+  description: Cron style schedule to repeat the action on.
+  returned: when I(state=present)
+  type: str
+  sample: '40 22 * * 1-5'
+min_size:
+  description: ASG min capacity.
+  returned: when I(state=present)
+  type: int
+  sample: 1
+max_size:
+  description: ASG max capacity.
+  returned: when I(state=present)
+  type: int
+  sample: 2
+desired_capacity:
+  description: ASG desired capacity.
+  returned: when I(state=present)
+  type: int
+  sample: 1
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+try:
+    from dateutil.parser import parse as timedate_parse
+    HAS_DATEUTIL = True
+except ImportError:
+    HAS_DATEUTIL = False
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+
+
+def format_request():
+    params = dict(
+        AutoScalingGroupName=module.params.get('autoscaling_group_name'),
+        ScheduledActionName=module.params.get('scheduled_action_name'),
+        Recurrence=module.params.get('recurrence')
+    )
+
+    # Some of these params are optional
+    if module.params.get('desired_capacity') is not None:
+        params['DesiredCapacity'] = module.params.get('desired_capacity')
+
+    if module.params.get('min_size') is not None:
+        params['MinSize'] = module.params.get('min_size')
+
+    if module.params.get('max_size') is not None:
+        params['MaxSize'] = module.params.get('max_size')
+
+    if module.params.get('time_zone') is not None:
+        params['TimeZone'] = module.params.get('time_zone')
```

Comment:
This fails for me:
```
Unknown parameter in input: "TimeZone", must be one of: AutoScalingGroupName, ScheduledActionName, Time, StartTime, EndTime, Recurrence, MinSize, MaxSize, DesiredCapacity
```
Though I see it in the boto docs.  Is a newer version of botocore required for this module than the collection minimum? https://github.com/ansible-collections/community.aws/blob/main/requirements.txt  
If it is then we need to document that in the module docs for the affected parameter and do a check for it, like in this example https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/aws_msk_cluster.py#L580

Reply:
Yes I think it's quite a recent addition, Makes sense I'll add the check. Thanks for the pointer 🙏

Reply:
Added in botocore 1.20.24  (Tue Mar 9 19:45:18 2021)